### PR TITLE
refactor(radio): rewrite with hook

### PR DESCRIPTION
**Pull request by hengkx**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  refactor Radio with hook         |
| 🇨🇳 Chinese | Radio 用 react hook 重构         |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️ 

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/radio/__tests__/group.test.js [hunk 1]
```diff
@@ -187,17 +160,17 @@ describe('Radio Group', () => {
...
     [undefined, null].forEach(newValue => {
       it(`should set value back when value change back to ${newValue}`, () => {
         const options = [{ label: 'Bamboo', value: 'bamboo' }];
         const wrapper = mount(<RadioGroup value="bamboo" options={options} />);
-        expect(wrapper.state().value).toEqual('bamboo');
-
+        expect(wrapper.find('.ant-radio-checked > input').prop('value')).toEqual('bamboo');
         wrapper.setProps({ value: newValue });
-        expect(wrapper.state().value).toEqual(newValue);
+        console.log(wrapper.find('.ant-radio-checked').html());
```
Comment (hengkx) [inline components/radio/__tests__/group.test.js:172]:
```
<span class="ant-radio"><input type="radio" class="ant-radio-input" value="bamboo" checked=""><span class="ant-radio-inner"></span></span>
```
明明没有`class`为什么还可以被查找到

**Comment by afc163** [comment]
在 https://github.com/ant-design/ant-design/pull/24480 基础上重新搞下吧。

File: components/radio/__tests__/group.test.js (hunk 1 above)
Comment (hengkx) [reply components/radio/__tests__/group.test.js:172]: @afc163 @xrkffgg 眼睛都花了，哪写的不对。。。

Comment (hengkx) [reply components/radio/__tests__/group.test.js:172]: 加个 wrapper.update()

**Comment by hengkx** [comment]
#24485

**Pull request closed without merging**